Maven commands fail randomly in CI/CD workflow

Latest error thrown:

```console
Error:  Error executing Maven.
java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1301)
	at java.base/java.util.Properties.setProperty(Properties.java:229)
	at org.apache.maven.cli.MavenCli.populateProperties(MavenCli.java:1656)
	at org.apache.maven.cli.MavenCli.properties(MavenCli.java:612)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:282)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:47)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:156)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:72)
> nx run shared-java-challenge-util:build
Executing command: ./mvnw package 
Failed to execute command: ./mvnw package 
Error: Command failed: ./mvnw package 
    at checkExecSyncError (node:child_process:828:11)
    at execSync (node:child_process:899:15)
    at runBuilderCommand (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/common/src/lib/core/jvm/utils.js:19:38)
    at runBootPluginCommand (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/utils/boot-utils.js:15:43)
    at /__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:10:62
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:113:16)
    at buildExecutor (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:8:20)
Error
    at /__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:12:19
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:113:16)
    at buildExecutor (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:8:20)
    at /__w/challenge-registry/challenge-registry/node_modules/@nrwl/tao/src/commands/run.js:147:23
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at __awaiter (/__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:113:16)
    at runExecutorInternal (/__w/challenge-registry/challenge-registry/node_modules/@nrwl/tao/src/commands/run.js:127:34)
    at Object.<anonymous> (/__w/challenge-registry/challenge-registry/node_modules/@nrwl/tao/src/commands/run.js:219:54)
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
```

Another instance of the issue at hand that was solved by a second run:

```console
 Warning: JAVA_HOME environment variable is not set.
Warning: JAVA_HOME environment variable is not set.
Warning: JAVA_HOME environment variable is not set.
Could not set executable permissions for: /root/.m2/wrapper/dists/apache-maven-3.8.5-bin/5i5jha092a3i37g0paqnfr15e0/apache-maven-3.8.5/bin/mvn
Please do this manually if you want to use Maven.
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:47)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:156)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:72)
Caused by: java.lang.NoClassDefFoundError: org/apache/maven/exception/ExceptionHandler
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3402)
	at java.base/java.lang.Class.getMethodsRecursive(Class.java:3543)
	at java.base/java.lang.Class.getMethod0(Class.java:3529)
	at java.base/java.lang.Class.getMethod(Class.java:2225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.getEnhancedMainMethod(Launcher.java:168)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:261)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	... 7 more
Caused by: java.lang.ClassNotFoundException: org.apache.maven.exception.ExceptionHandler
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	... 17 more

> nx run challenge-config-service:build

Executing command: ./mvnw package 
Failed to execute command: ./mvnw package 
Error: Command failed: ./mvnw package 
    at checkExecSyncError (node:child_process:828:11)
    at execSync (node:child_process:899:15)
    at runBuilderCommand (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/common/src/lib/core/jvm/utils.js:19:38)
    at runBootPluginCommand (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/utils/boot-utils.js:15:43)
    at /__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:10:62
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:114:16)
    at buildExecutor (/__w/challenge-registry/challenge-registry/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:8:20)
    at /__w/challenge-registry/challenge-registry/node_modules/nx/src/command-line/run.js:101:23
    at Generator.next (<anonymous>)
    at /__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/__w/challenge-registry/challenge-registry/node_modules/tslib/tslib.js:114:16)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -53,13 +53,13 @@ jobs:
       - name: Install Node.js dependencies
         if: steps.cache-node.outputs.cache-hit != 'true'
         run: yarn install --frozen-lockfile
-      - name: Prepare the projects
-        run: yarn nx run-many --all --parallel --target=prepare
-      - name: Setup Python virtualenvs
-        if: steps.cache-pipenv.outputs.cache-hit != 'true'
-        run: yarn nx python api
-      - run: yarn nx affected --target=lint --parallel --max-parallel=3
-      - run: yarn nx affected --target=build --parallel --max-parallel=3
+      # - name: Setup Python virtualenvs
+      #   if: steps.cache-pipenv.outputs.cache-hit != 'true'
+      #   run: yarn nx python api
+      - run: yarn nx affected --target=prepare
+      - run: yarn nx affected --target=install-maven --parallel=1
+      - run: yarn nx affected --target=lint
+      - run: yarn nx affected --target=build
       - run: yarn nx run-many --all --target=test --parallel --max-parallel=2
       - name: Merge coverage reports
         run: yarn coverage:merge
@@ -88,12 +88,12 @@ jobs:
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
-      - name: Setup pipenv cache
-        id: cache-pipenv
-        uses: actions/cache@v3
-        with:
-          path: ~/.local/share/virtualenvs
-          key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+      # - name: Setup pipenv cache
+      #   id: cache-pipenv
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: ~/.local/share/virtualenvs
+      #     key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
       - name: Setup Maven cache
         uses: actions/cache@v3
         with:
@@ -106,11 +106,11 @@ jobs:
       - name: Install Node.js dependencies
         if: steps.cache-node.outputs.cache-hit != 'true'
         run: yarn install --frozen-lockfile
-      - name: Prepare the projects
-        run: yarn nx run-many --all --parallel --target=prepare
-      - name: Setup Python virtualenvs
-        if: steps.cache-pipenv.outputs.cache-hit != 'true'
-        run: yarn nx python api
-      - run: yarn nx affected --target=lint --parallel --max-parallel=3
-      - run: yarn nx affected --target=build --parallel --max-parallel=3
-      - run: yarn nx affected --target=test --parallel --max-parallel=2
+      # - name: Setup Python virtualenvs
+      #   if: steps.cache-pipenv.outputs.cache-hit != 'true'
+      #   run: yarn nx python api
+      - run: yarn nx affected --target=prepare
+      - run: yarn nx affected --target=install-maven --parallel=1
+      - run: yarn nx affected --target=lint
+      - run: yarn nx affected --target=build
+      - run: yarn nx affected --target=test --parallel=2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,10 +57,11 @@ jobs:
       #   if: steps.cache-pipenv.outputs.cache-hit != 'true'
       #   run: yarn nx python api
       - run: yarn nx affected --target=prepare
-      - run: yarn nx affected --target=install-maven --parallel=1
+      - run: yarn nx affected --target=prepare-java --parallel=1
       - run: yarn nx affected --target=lint
       - run: yarn nx affected --target=build
-      - run: yarn nx run-many --all --target=test --parallel --max-parallel=2
+      - run: yarn nx affected --target=test
+      # - run: yarn nx run-many --all --target=test --parallel --max-parallel=2
       - name: Merge coverage reports
         run: yarn coverage:merge
       - name: Push coverage report to Coveralls
@@ -110,7 +111,7 @@ jobs:
       #   if: steps.cache-pipenv.outputs.cache-hit != 'true'
       #   run: yarn nx python api
       - run: yarn nx affected --target=prepare
-      - run: yarn nx affected --target=install-maven --parallel=1
+      - run: yarn nx affected --target=prepare-java --parallel=1
       - run: yarn nx affected --target=lint
       - run: yarn nx affected --target=build
-      - run: yarn nx affected --target=test --parallel=2
+      - run: yarn nx affected --target=test

--- a/apps/challenge-api-gateway/project.json
+++ b/apps/challenge-api-gateway/project.json
@@ -10,6 +10,15 @@
         "cwd": "apps/challenge-api-gateway"
       }
     },
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-api-gateway"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/apps/challenge-api-gateway/project.json
+++ b/apps/challenge-api-gateway/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-api-gateway"
       }

--- a/apps/challenge-api-gateway/project.json
+++ b/apps/challenge-api-gateway/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-api-gateway"
       }

--- a/apps/challenge-api-gateway/project.json
+++ b/apps/challenge-api-gateway/project.json
@@ -10,11 +10,11 @@
         "cwd": "apps/challenge-api-gateway"
       }
     },
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-api-gateway"
       }

--- a/apps/challenge-api-gateway/project.json
+++ b/apps/challenge-api-gateway/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-api-gateway"
       }

--- a/apps/challenge-auth-service/project.json
+++ b/apps/challenge-auth-service/project.json
@@ -10,11 +10,11 @@
         "cwd": "apps/challenge-auth-service"
       }
     },
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-auth-service"
       }

--- a/apps/challenge-auth-service/project.json
+++ b/apps/challenge-auth-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-auth-service"
       }

--- a/apps/challenge-auth-service/project.json
+++ b/apps/challenge-auth-service/project.json
@@ -10,6 +10,15 @@
         "cwd": "apps/challenge-auth-service"
       }
     },
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-auth-service"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/apps/challenge-auth-service/project.json
+++ b/apps/challenge-auth-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-auth-service"
       }

--- a/apps/challenge-auth-service/project.json
+++ b/apps/challenge-auth-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-auth-service"
       }

--- a/apps/challenge-config-service/project.json
+++ b/apps/challenge-config-service/project.json
@@ -3,6 +3,15 @@
   "sourceRoot": "apps/challenge-config-service/src",
   "projectType": "application",
   "targets": {
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-config-service"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/apps/challenge-config-service/project.json
+++ b/apps/challenge-config-service/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-config-service"
       }

--- a/apps/challenge-config-service/project.json
+++ b/apps/challenge-config-service/project.json
@@ -3,11 +3,11 @@
   "sourceRoot": "apps/challenge-config-service/src",
   "projectType": "application",
   "targets": {
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-config-service"
       }

--- a/apps/challenge-config-service/project.json
+++ b/apps/challenge-config-service/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-config-service"
       }

--- a/apps/challenge-config-service/project.json
+++ b/apps/challenge-config-service/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-config-service"
       }

--- a/apps/challenge-core-service/pom.xml
+++ b/apps/challenge-core-service/pom.xml
@@ -102,6 +102,23 @@
 					</excludes>
 				</configuration>
 			</plugin>
+      <!-- <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>3.3.0</version>
+        <executions>
+          <execution>
+            <phase>dependency</phase>
+            <goals>
+              <goal>go-offline</goal>
+            </goals>
+            <configuration>
+              <excludeGroupIds>org.sagebionetworks.challenge</excludeGroupIds>
+              <excludeArtifactIds>challenge-util</excludeArtifactIds>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin> -->
 		</plugins>
     <pluginManagement>
       <plugins>

--- a/apps/challenge-core-service/project.json
+++ b/apps/challenge-core-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-core-service"
       }

--- a/apps/challenge-core-service/project.json
+++ b/apps/challenge-core-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-core-service"
       }

--- a/apps/challenge-core-service/project.json
+++ b/apps/challenge-core-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-core-service"
       }

--- a/apps/challenge-core-service/project.json
+++ b/apps/challenge-core-service/project.json
@@ -10,6 +10,15 @@
         "cwd": "apps/challenge-core-service"
       }
     },
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-core-service"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/apps/challenge-core-service/project.json
+++ b/apps/challenge-core-service/project.json
@@ -10,11 +10,11 @@
         "cwd": "apps/challenge-core-service"
       }
     },
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-core-service"
       }

--- a/apps/challenge-service-registry/project.json
+++ b/apps/challenge-service-registry/project.json
@@ -10,11 +10,11 @@
         "cwd": "apps/challenge-service-registry"
       }
     },
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-service-registry"
       }

--- a/apps/challenge-service-registry/project.json
+++ b/apps/challenge-service-registry/project.json
@@ -10,6 +10,15 @@
         "cwd": "apps/challenge-service-registry"
       }
     },
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-service-registry"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/apps/challenge-service-registry/project.json
+++ b/apps/challenge-service-registry/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-service-registry"
       }

--- a/apps/challenge-service-registry/project.json
+++ b/apps/challenge-service-registry/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-service-registry"
       }

--- a/apps/challenge-service-registry/project.json
+++ b/apps/challenge-service-registry/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-service-registry"
       }

--- a/apps/challenge-user-service/project.json
+++ b/apps/challenge-user-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-user-service"
       }

--- a/apps/challenge-user-service/project.json
+++ b/apps/challenge-user-service/project.json
@@ -10,11 +10,11 @@
         "cwd": "apps/challenge-user-service"
       }
     },
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "apps/challenge-user-service"
       }

--- a/apps/challenge-user-service/project.json
+++ b/apps/challenge-user-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "apps/challenge-user-service"
       }

--- a/apps/challenge-user-service/project.json
+++ b/apps/challenge-user-service/project.json
@@ -14,7 +14,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "apps/challenge-user-service"
       }

--- a/apps/challenge-user-service/project.json
+++ b/apps/challenge-user-service/project.json
@@ -10,6 +10,15 @@
         "cwd": "apps/challenge-user-service"
       }
     },
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "apps/challenge-user-service"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/libs/shared-java/challenge-util/project.json
+++ b/libs/shared-java/challenge-util/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve"
+          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "libs/shared-java/challenge-util"
       }

--- a/libs/shared-java/challenge-util/project.json
+++ b/libs/shared-java/challenge-util/project.json
@@ -3,6 +3,15 @@
   "sourceRoot": "libs/shared-java/challenge-util/src",
   "projectType": "library",
   "targets": {
+    "install-maven": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "commands": [
+          "./mvnw --version"
+        ],
+        "cwd": "libs/shared-java/challenge-util"
+      }
+    },
     "build": {
       "executor": "@nxrocks/nx-spring-boot:build",
       "options": {

--- a/libs/shared-java/challenge-util/project.json
+++ b/libs/shared-java/challenge-util/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:resolve -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
         ],
         "cwd": "libs/shared-java/challenge-util"
       }

--- a/libs/shared-java/challenge-util/project.json
+++ b/libs/shared-java/challenge-util/project.json
@@ -7,7 +7,7 @@
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge"
+          "./mvnw dependency:go-offline -DexcludeGroupIds=org.sagebionetworks.challenge || true"
         ],
         "cwd": "libs/shared-java/challenge-util"
       }

--- a/libs/shared-java/challenge-util/project.json
+++ b/libs/shared-java/challenge-util/project.json
@@ -3,11 +3,11 @@
   "sourceRoot": "libs/shared-java/challenge-util/src",
   "projectType": "library",
   "targets": {
-    "install-maven": {
+    "prepare-java": {
       "executor": "@nrwl/workspace:run-commands",
       "options": {
         "commands": [
-          "./mvnw --version"
+          "./mvnw dependency:resolve"
         ],
         "cwd": "libs/shared-java/challenge-util"
       }

